src/bin/ipa_bench/gen_events.rs

```diff
@@ -0,0 +1,481 @@
+use super::sample::Sample;
+use log::{debug, info, trace};
+use rand::SeedableRng;
+use rand::{CryptoRng, Rng, RngCore};
+use rand_chacha::ChaCha20Rng;
+use rand_distr::num_traits::ToPrimitive;
+use rand_distr::{Bernoulli, Distribution};
+use raw_ipa::helpers::models::{
+    Event as EEvent, SecretSharable, SecretShare, SourceEvent as ESourceEvent,
+    TriggerEvent as ETriggerEvent,
+};
+use serde::{Deserialize, Serialize};
+use std::io;
+use std::time::Duration;
+
+const DAYS_IN_EPOCH: u64 = 7;
+
+#[derive(Clone)]
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+pub struct Event {
+    // For this tool, we'll fix the length of a matchkey to u64
+    pub matchkeys: Vec<u64>,
+    pub epoch: u8,
+    pub timestamp: u32,
+}
+
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+pub struct SourceEvent {
+    pub event: Event,
+    pub breakdown_key: String,
+}
+
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+pub struct TriggerEvent {
+    pub event: Event,
+    pub value: u32,
+    pub zkp: String,
+}
+
+#[cfg_attr(feature = "enable-serde", derive(Serialize, Deserialize))]
+pub enum EventType {
+    // Source event in clear
+    S(SourceEvent),
+    // Trigger event in clear
+    T(TriggerEvent),
+    // Source event in cipher if --secret-share option is enabled
+    ES(ESourceEvent),
+    // Trigger event in cipher if --secret-share option is enabled
+    ET(ETriggerEvent),
+}
```

Comment:
I think that you want Event, and EncryptedEvent as discrete types.  Also, this isn't a *Type enum as it contains details of the event, not just the type.

Ultimately, there might be three values here:
1. The raw event, with values included directly.
2. A number of secret-shared events.
3. Encrypted secret-shared events.

Having explicit types for each is probably best.

This tool can concentrate on the first step: generating synthetic values for (1).